tests/end_to_end/features/steps/serving.py

```diff
@@ -190,6 +204,17 @@ def step_impl(context):
         i += 1
 
 
+@then("I can get a list containing the correct number of features")
+def step_impl(context):
+    i = 0
+    for entity, features in context.iter:
+        if i >= len(context.expected):
+            break
```

Comment:
should probably throw

Reply:
The counter will always exceed the expected because the stream will have more results than we care to test for. This is meant to avoid a long running test.